frontend/src/pages/ProgressionChecker/ProgressionChecker.jsx

```diff
@@ -38,31 +39,36 @@ const degreeData = {
 };
 
 const ProgressionChecker = () => {
-  const [degree, setDegree] = useState({});
   const [isLoading, setIsLoading] = useState(true);
-  const [courses, setCourses] = useState({});
 
-  const fetchDegree = async () => {
-    setDegree(degreeData);
-    const res = await axios.get(
-      `/programs/getStructure/3778/${degreeData.concentrations[0].code}`,
-    );
+  const {
+    programCode, specialisation, minor,
+  } = useSelector((state) => state.degree);
 
-    setCourses(res.data.structure);
-    setIsLoading(false);
-  };
+  const [structure, setStructure] = useState({});
 
   useEffect(() => {
-    fetchDegree();
-  }, []);
+    // get structure of degree
+    const fetchStructure = async () => {
+      try {
+        const res = await axios.get(`/programs/getStructure/${programCode}/${specialisation}${minor && `/${minor}`}`);
```

Comment:
could do a ternary for the url to use beforehand to avoid the nesting just to make it more readable